src/core/schema/src/main/python/TypeConversionUtils.py

```diff
@@ -28,16 +30,20 @@ def complexTypeConverter(name, value, cache):
         _java_obj:
     """
     cache[name]=value
-    if isinstance(value, list):
-        java_value=[]
-        for v in value:
-            if hasattr(v, "_transfer_params_to_java"):
-                v._transfer_params_to_java()
-            java_value.append(v._java_obj)
-        return java_value
-    if hasattr(value, "_transfer_params_to_java"):
-        value._transfer_params_to_java()
-    if hasattr(value, "_java_obj"):
-        return value._java_obj
-    else:
-        return value._to_java()
+    try:
+        if isinstance(value, list):
```

Comment:
You are probably right, I just wanted to keep any alterations to this code to a minimum for fear of inadvertently breaking something. After whacking through this jungle, I think we need to refactor codegen and how we abstract the python-scala handshaking. It should probabbly be planned out and discussed. However this is a fix for now.